### PR TITLE
Get Single LTK Endpoint

### PR DESCRIPTION
**Pull request by amagana3**
This PR adds a new endpoint to retrieve a single LTK user from the current account with account information inferred from STS credentials. This is working in both the tests as well as my local test suite.

Also cleaned up another test to remove redundant variable which Go's linter was complaining about. Did not affect the test results, simply cleaned up the code.

### CODE REVIEW

File: iam_ltk.go [hunk 1]
```diff
@@ -21,6 +21,12 @@ type GetLongTermKeysResponse struct {
 	LongTermKeys []LongTermKey `json:"longTermKeys"`
 }
 
+// GetLongTermKeyResponse is used to represent a single long term key.
+type GetLongTermKeyResponse struct {
+	BaseResponse
+	LongTermKey `json:"longTermKey"`
```
Comment (webbbarker) [inline iam_ltk.go:28]: Careful.  This is called `embedding` and actually makes your `GetLongTermKeyResponse` a subclass of both BaseResponse (which is true) and `LongTermKey` which is kinda weird?  You probably want this to be a named field unless your intent is to have users see this response as having the bare long term key fields available AND you reuse the collection of those fields in other places in the codebase as a unit.  What I am trying to get at is that this is not very idiomatic `golang`, so we need to have a good reason for it to be that way or follow the more traditional approach of naming this field (probably also named `LongTermKey`).

Comment (webbbarker) [reply iam_ltk.go:28]: Reading more of the file, maybe this is OK as is.  Leaving the note so we can discuss.

File: iam_ltk.go [hunk 2]
```diff
@@ -100,6 +106,37 @@ func (c *Client) GetLongTermKeys() (*GetLongTermKeysResponse, error) {
 	return cr, nil
 }
 
+// GetLongTermKey gets a single LTK for an account
+// If no error is returned, then you will receive an LTK for the given account.
+func (c *Client) GetLongTermKey(iamUsername string) (*GetLongTermKeyResponse, error) {
+	log.Printf("[INFO] Getting long term key")
+
+	req, err := c.NewRequest(nil, "GET", "/ltk/search/"+iamUsername)
```
Comment (webbbarker) [inline iam_ltk.go]: So I didn't get to review you ALKS Core PR, but how is the account name being communicated?  Right now this endpoint looks like it searches every account for a matching LTK.

**Review by webbbarker — CHANGES_REQUESTED** [review]
Couple comments, let's discuss.

**Comment by amagana3** [comment]
### Update
So I've modified the GET LTK endpoint so it checks for whether we're using STS or non-STS credentials and calls the appropriate endpoint as this endpoint is not _just_ for one or the other.

Also, the `DELETE` and `CREATE` endpoints required no modification because @webbbarker, your previous PR (modifying the login) helped remediate the issue I was having. Now whenever a user calls alks-go their account information is populated when using STS credentials. 👍 

This PR is now up to date and working without issues!

File: iam_ltk.go [hunk 3]
```diff
@@ -111,7 +112,25 @@ func (c *Client) GetLongTermKeys() (*GetLongTermKeysResponse, error) {
...
+
+	accountID, err := c.AccountDetails.GetAccountNumber()
+	if err != nil {
+		return nil, fmt.Errorf("error reading Account value: %s", err)
+	}
+
+	roleName, err := c.AccountDetails.GetRoleName(false)
+	if err != nil {
+		return nil, fmt.Errorf("error reading Role value: %s", err)
+	}
```
Comment (webbbarker) [inline iam_ltk.go]: You should only call these two methods if we aren't using STS credentials.  _Technically_ they might not be populated.

**Review by webbbarker — CHANGES_REQUESTED** [review]
Let's rearrange the calls a little bit to avoid potential errors.  Otherwise looks good.

**Review by webbbarker — APPROVED** [review]
LGTM.  👍

**Pull request merged**